### PR TITLE
Bump version number to 8.18 and add compat infrastructure.

### PR DESCRIPTION
**Pull request by Zimmi48**
Part of the commit was auto-generated by running `dev/tools/update-compat.py --master`.
Can I get an assignee so that this is merged quickly when CI passes?

### CODE REVIEW

File: tools/configure/configure.ml [hunk 1]
```diff
@@ -22,8 +22,8 @@ open CmdArgs.Prefs
 
 let (/) = Filename.concat
 
-let coq_version = "8.17+alpha"
-let vo_magic = 81699
+let coq_version = "8.18+alpha"
+let vo_magic = 81899
```
Comment (SkySkimmer) [inline tools/configure/configure.ml:26]: Why's it increased by 200?

Comment (Zimmi48) [reply tools/configure/configure.ml:26]: Well spotted, thanks!

**Comment by Zimmi48** [comment]
Not re-running a full CI now because the previous run has already caught that fiat-parsers needs to be fixed (cc @JasonGross).

BTW, fiat-crypto gives an anomaly now (did we break it in master?).

**Comment by Zimmi48** [comment]
@JasonGross Please fix fiat-parser compatibility ASAP.

**Comment by SkySkimmer** [comment]
Let's see 
@coqbot run full ci

**Comment by Zimmi48** [comment]
@SkySkimmer seems like CI has passed besides already known failures

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Pull request merged**